providers/base/api/handlers/api_query_handler.go

```diff
@@ -198,17 +204,37 @@ func (h *APIQueryHandlerImpl[K, V]) subTask(
 
 		resp, err := h.requestHandler.Do(ctx, url)
 		if err != nil {
-			h.writeResponse(responseCh, providertypes.NewGetResponseWithErr[K, V](ids, errors.ErrDoRequestWithErr(err)))
+			status := providertypes.ErrorUnknown
+			if resp != nil {
+				status = providertypes.ErrorCode(resp.StatusCode)
+			}
+
+			h.writeResponse(responseCh, providertypes.NewGetResponseWithErr[K, V](ids,
+				providertypes.NewErrorWithCode(
+					errors.ErrDoRequestWithErr(err),
+					status,
+				)),
+			)
```

Comment:
a bit of a footgun here is that we return a separate codes for http rate limits and the provider error code. Im pretty sure the request handler will return an error anytime it doesnt get a 200, so its likely we never even reach the code below. tldr, i think just having the http status code included in sufficient.

Reply:
not critical since we can change this but im curious to hear your thoughts.

Reply:
the error returned from the API handler will include the code right?